Src/IronPython/Runtime/Bytes.cs

```diff
@@ -30,7 +30,18 @@ public class Bytes : IList<byte>, IReadOnlyList<byte>, IEquatable<Bytes>, ICodeF
             _bytes = new byte[0];
         }
 
-        public Bytes(object? source) : this(ByteArray.GetBytes(source, useHint: true)) { }
+        public Bytes(CodeContext context, object? source) {
+            if (PythonOps.TryGetBoundAttr(context, source, "__bytes__", out object? func)) {
```

Comment:
This should probably use `PythonTypeOps.TryInvokeUnaryMember`. See https://github.com/IronLanguages/ironpython3/issues/790. Otherwise the following (minus any typos) would work when it shouldn't.
```Python
class test: pass
t = test()
t.__bytes__ = lambda x: b"1"
bytes(t)
```

Reply:
Hmm, I thought it should, but it shouldn't indeed. Funny, it still doesn't work, and with a strange error message:
````
TypeError: <lambda$175>() takes exactly 1 argument (1 given)
````

Reply:
That would be a typo on my end (writing code on a phone usually doesn't work out well), should have been `t.__bytes__ = lambda: b"1"`.

Reply:
But shoudn't the message read `<lambda$175>() takes exactly 1 argument (0 given)`?

Reply:
Yes, the message is indeed messed up. I think it has to do with it counting "self" as an argument? Not the first time I've seen messed up argument counts but never bothered to investigate.